assets/src/components/kubernetes/common/ResourceDetails.tsx

```diff
@@ -36,6 +46,33 @@ export default function ResourceDetails({
   const currentTab = tabs.find(({ path }) => path === (tab ?? ''))
   const [headerContent, setHeaderContent] = useState<ReactNode>()
   const pageHeaderContext = useMemo(() => ({ setHeaderContent }), [])
+  const { clusterId, name, namespace, crd } = useParams()
+  const kindPathMatch = useMatch(`${getKubernetesAbsPath(clusterId)}/:kind/*`)
+  const kind = useMemo(
+    () => crd ?? pluralize(kindPathMatch?.params?.kind || '', 1),
+    [crd, kindPathMatch?.params?.kind]
+  )
+
+  const resourceQuery = useMemo(
+    () => (namespace ? useNamespacedResourceQuery : useResourceQuery),
+    [namespace]
+  )
+
+  const { data } = resourceQuery({
+    client: KubernetesClient(clusterId ?? ''),
+    skip: !clusterId || kind === 'secret',
+    fetchPolicy: 'no-cache',
+    variables: { kind, name, namespace } as ResourceQueryVariables &
+      NamespacedResourceQueryVariables,
+  })
+
+  const prompt = useMemo(() => {
+    return data?.handleGetResource?.Object
+      ? `Describe the following Kubernetes ${kind} resource: ${JSON.stringify(data?.handleGetResource?.Object)}`
```

Comment:
Just double-checking, the capitalization is correct here?

Reply:
It's not actually. Do we need it?

Reply:
oh i mean is there a field called `Object` instead of `object`?